invisible_cities/detsim/simulate_S1.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import pandas as pd
+
+from typing import Tuple
+from typing import Callable
+
+from scipy.optimize import brentq
+
+from .. core.core_functions import in_range
+from .. core import system_of_units as units
+
+from .. io.dst_io  import load_dst
+
+
+def compute_S1_pes_at_pmts(xs      : np.ndarray,
+                           ys      : np.ndarray,
+                           zs      : np.ndarray,
+                           photons : np.ndarray,
+                           LT      : Callable  )->np.ndarray:
+    """Compute the pes generated in each PMT from S1 photons
+    Parameters:
+        :LT: function
+            The Light Table in functional form
+        :photons: np.ndarray
+            The photons generated at each hit
+        :xs, ys, zs: np.ndarray
+            hit position
+    Returns:
+        :pes: np.ndarray
+            photoelectrons at each PMT produced by all hits.
+    """
+    pes = photons[:, np.newaxis] * LT(xs, ys, zs)
+    pes = np.random.poisson(pes)
+    return pes.T
+
+tau1 = 4.5; c1 = 0.1
+tau2 = 100; c2 = 0.9
+
+N = 1/(c1*tau1 + c2*tau2)
+A1 = tau1*c1*N
+A2 = tau2*c2*N
+
+def S1_minimize(x, P):
+    return A1*np.exp(-x/tau1) + A2*np.exp(-x/tau2) - P
+
+def generate_S1_time(size : int)->np.ndarray:
+    """Generates random numbers following the hyper-exponential
+    distribution: 0.9 exp(-x/4.5) + 0.1 exp(-x/100).
+    Parameters:
+        :size: int
+            number of random numbers
+    Returns:
+        :sol: np.ndarray
+            the random number values
+    """
+    sol = []
+    for i in range(size):
+        P = np.random.random()
+        sol.append(brentq(S1_minimize, 0, 10000, args=P))
+    return np.array(sol)
+
+
+def generate_S1_times_from_pes(S1_pes_at_pmts : np.ndarray,
+                               hit_times      : np.ndarray)->list:
+    """Given the S1_pes_at_pmts, this function returns the times at which the pes
+    are be distributed (see generate_S1_time function).
+    It returns a list whose elements are the times at which the photoelectrons in that PMT
+    are generated.
+    Parameters:
+        :S1_pes_at_pmts: np.ndarray
+            the pes at each PMT generated by each hit
+    Returns:
+        :S1_times: list[np.ndarray,..]
+            Each element are the S1 times for a PMT. If certain sensor
+            do not see any pes, the array is empty.
+    """
+    S1_times = []
+    for pes_at_pmt in S1_pes_at_pmts:
+        repeated_times = np.repeat(hit_times, pes_at_pmt)
+        times = generate_S1_time(np.sum(pes_at_pmt))
+        S1_times.append(times + repeated_times)
+    return S1_times
+
+
+def S1_waveforms_creator(s1_lighttable, ws, wf_pmt_bin_width):
+    S1_LT = create_lighttable_function(os.path.expandvars(s1_lighttable))
+
+    def create_S1_waveforms_from_hits(x, y, z, time, energy, tmin, buffer_length):
+        s1_photons     = np.random.poisson(energy / ws)
```

Comment:
Maybe this doesn't belong here, lets take it out in a 
```
def num_scintillation_photons (energy, ws):
    return np.random.poisson(energy, ws) 
```
with docstrings and type annotation of course :)